Client/Frontend/Shields/ShareShieldsController.swift

```diff
@@ -0,0 +1,280 @@
+        ).with {
+            $0.width += UX.contentSizeChange
+        }
+        
+        navigationController?.preferredContentSize = preferredContentSize
+    }
+        
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        
+        if #available(iOS 13.0, *) {
+            if traitCollection.userInterfaceStyle != previousTraitCollection?.userInterfaceStyle {
+                applyTheme(theme)
+            }
+        }
+    }
+    
+    // MARK: Internal
+    
+    private func doLayoutAndAddGestureRecognizers() {
+        let shareTypeViewEmail = ShareTypeView(type: .email)
+        let shareEmailGesture = UITapGestureRecognizer(target: self, action: #selector(shareEmailClicked))
+        shareTypeViewEmail.addGestureRecognizer(shareEmailGesture)
+        
+        let shareTypeViewTwitter = ShareTypeView(type: .twitter)
+        let shareTwitterGesture = UITapGestureRecognizer(target: self, action: #selector(shareTwitterClicked))
+        shareTypeViewTwitter.addGestureRecognizer(shareTwitterGesture)
+        
+        let shareTypeViewFacebook = ShareTypeView(type: .facebook)
+        let shareFacebookGesture = UITapGestureRecognizer(target: self, action: #selector(shareFacebookClicked))
+        shareTypeViewFacebook.addGestureRecognizer(shareFacebookGesture)
+        
+        let shareTypeViewDefault = ShareTypeView(type: .default)
+        let shareDefaultGesture = UITapGestureRecognizer(target: self, action: #selector(shareDefaultClicked))
+        shareTypeViewDefault.addGestureRecognizer(shareDefaultGesture)
+         
+        view.addSubview(contentStackView)
+
+        contentStackView.snp.makeConstraints {
+            $0.top.leading.trailing.equalTo(view.safeAreaLayoutGuide)
+        }
+
+        contentStackView.addStackViewItems(
+            .view(shareTypeViewEmail),
+            .view(shareTypeViewTwitter),
+            .view(shareTypeViewFacebook),
+            .view(shareTypeViewDefault)
+        )
+    }
+    
+    // MARK: Themeable
+    
+    func applyTheme(_ theme: Theme) {
+        view.backgroundColor = theme.isDark ? BraveUX.popoverDarkBackground : UIColor.white
+        
+        contentStackView.arrangedSubviews.forEach { subview in
+            if let typeView = subview as? ShareTypeView {
+                typeView.applyTheme(theme)
+            }
+        }
+    }
+    
+    // MARK: Actions
+    
+    @objc func shareEmailClicked(gesture: UITapGestureRecognizer) {
+        actionHandler?(.shareEmailClicked)
+    }
+    
+    @objc func shareTwitterClicked(gesture: UITapGestureRecognizer) {
+        actionHandler?(.shareTwitterClicked)
+    }
+    
+    @objc func shareFacebookClicked(gesture: UITapGestureRecognizer) {
+        actionHandler?(.shareFacebookClicked)
+    }
+    
+    @objc func shareDefaultClicked(gesture: UITapGestureRecognizer) {
+        actionHandler?(.shareDefaultClicked)
```

Comment:
Looks like a lot of code duplicated here. Is is possible to have one function and just pass the enum directly to the actionHandler?

Reply:
Yeap ofcourse we can do it